### PR TITLE
134/2033 Simulation Thumbnail

### PR DESCRIPTION
**Pull request by bilalesi**
- add new endpoint to generate the thumbnail based on the type (stimulus, simulation)
- the endpoint will fetch the simulation configuration file which has all the necessary data to generate the plots depends on the type

### CODE REVIEW

**Review by kplatis — COMMENTED** [review]
@bilalesi Thank you for the feature, I added a couple of comments. Not sure how urgent the deployment is but if its not super urgent can you add some tests? Also can you change the target branch to dev instead and squash the commits when merging?

File: api/services/simulation_img.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    if config.target == "stimulus":
+        stimulus_config = simulation_config.stimulus
+        # in the future the stimulus may have different configs
+        # we should agree how the user can specify the stimulus thumb needed
+        if len(stimulus_config) > 0:
+            data = stimulus_config[0].data
+    elif config.target == "simulation":
+        data = simulation_config.simulation
+
+    if len(data) > 0:
```
[the remark this answers is not in the document]
Comment (bilalesi) [reply api/services/simulation_img.py:45]: okey, i will make the changes

File: api/services/simulation_img.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    SimulationConfigurationFile,
+    SimulationGenerationInput,
+)
+from api.services.nexus import fetch_file_content
+
+
+def generate_simulation_plots(
+    access_token: str,
+    config: SimulationGenerationInput,
+):
```
[the remark this answers is not in the document]
Comment (bilalesi) [reply api/services/simulation_img.py:23]: sure 👍

File: api/models/common.py [hunk 3]
```diff
@@ -16,6 +16,64 @@ class ImageGenerationInput(BaseModel):
...
+class SingleNeuronModelSimulationConfig(BaseModel):
+    """
+    Simulation configuration (allow both single neuron and synaptome)
+    """
+
+    celsius: float
+    hypamp: float
+    vinit: float
+    injectTo: str
+    recordFrom: Any
```
[the remark this answers is not in the document]
Comment (bilalesi) [reply api/models/common.py:42]: all this values are unnecessary for the API, I just added them to be able to use right data in the endpoint implementation

**Comment by kplatis** [comment]
@bilalesi After rebasing, can you also update the `CHANGELOG` under v0.6.0 with the feature you implemented?

File: tests/test_services/test_morpho_img.py [hunk 4]
```diff
@@ -7,7 +7,6 @@
 from unittest.mock import patch
 from api.services.morpho_img import generate_morphology_image
 from tests.fixtures.utils import load_content
-from tests.fixtures.nexus import morphology_content_url, access_token
```
Comment (kplatis) [inline tests/test_services/test_morpho_img.py:10]: @bilalesi I think this removal breaks the tests

Comment (bilalesi) [reply tests/test_services/test_morpho_img.py]: fixed

**Review by kplatis — APPROVED** [review]

**Pull request merged**